docs/modules/SDDs/pages/0022-managed-services-in-cluster.adoc

```diff
@@ -0,0 +1,111 @@
+
+This SDD doesn't define how this service instance object is handled and managed. It could come together with the application deployment objects of the customer but also being stored in a separate GitOps repository just for these objects.
+
+A Kubernets Operator could also be responsible to manage different kinds of application services, it isn't strictly necessary that each Operator is only responsible for one service.
+
+Each provided service will be documented in the Commodore Component which installs the Kubernetes Operator for the service.
+
+=== User Stories
+
+==== Kafka with Strimzi Operator - Application in Cluster
+
+_As the user of the platform I want to use Apache Kafka for my application runnign in the same Kubernetes cluster as a managed service._
+
+* Project Syn Commodore Component `component-kafka-strimzi` deploys Strimzi to the cluster and configures it according to best practices.
+* The user of the platform creates a `Kafka` object and puts in parameters as needed. The object is stored at the same place as the other application deployment objects and also deployed to the cluster with the same mechanisms (f.e. GitOps).
+* The Strimzi Kubernetes Operator deploys Kafka and creates a secret with the connection parameters for the users application to consume.
+* Via the monitoring and backup configuration deployed by the Commodore Component the application service is automatically monitored and backed up.
+* For keeping the Strimzi Kubernetes Operator up-to-date the Project Syn mechanisms are used.
+* For keeping Kafka up-to-date the user of the platform has to maintain the `Kafka` object.
+
+==== Kafka with Strimzi Operator - Application outside Cluster
+
+_As an application owner I want to use Apache Kafka from my application running outside of the cluster._
+
+Similar to first use case, but:
+
+* The `Kafka` object is stored in _a_ GitOps repository and deployed with the Project Syn provided Argo CD.
+* The connection parameters are handed over to the application owner.
+* To access Kafka the Kubernetes services are configured in a way so that Kafka is reachable from outside of the cluster.
+
+=== Implementation Details/Notes/Constraints
+
+* It is possible that multiple Kubernetes Operator implementations exists for an application service. It is perfectly possible that the user could choose between several Kubernetes Operators for the same application service - depending on the customers needs.
+* In the case of having a Kubernetes Cluster which only provides an application service, a central GitOps repository can be used which stores all application instance objects. As the user of the application service doesn't get access to the cluster, there is no way of tampering with the Argo CD configuration.
+** An alternative could be to create a GitOps repository per customer which contains all the service instance objects and allow the customer to do some kind of self-service.
```

Comment:
A repository per customer whcih then again contains everything? I might misread that and it might be worth to improve on the wording.

Reply:
Will improve that.